src/app/fdctl/run/tiles/fd_sign.c

```diff
@@ -111,57 +103,48 @@ after_frag( void *             _ctx,
             int *              opt_filter,
             fd_mux_context_t * mux ) {
   (void)seq;
-  (void)opt_sig;
   (void)opt_chunk;
-  (void)opt_sz;
   (void)opt_tsorig;
   (void)opt_filter;
   (void)mux;
 
+  ulong sz        = *opt_sz;
+  ulong sig       = *opt_sig;
+  int   sign_type = (int)(uint)sig;
+
   fd_sign_ctx_t * ctx = (fd_sign_ctx_t *)_ctx;
+  int role = ctx->in_role[ in_idx ];
 
   FD_TEST( in_idx<MAX_IN );
 
-  switch( ctx->in_role[ in_idx ] ) {
-    case FD_KEYGUARD_ROLE_LEADER: {
-      if( FD_UNLIKELY( !fd_keyguard_payload_authorize( ctx->_data, 32UL, FD_KEYGUARD_ROLE_LEADER ) ) ) {
-        FD_LOG_EMERG(( "fd_keyguard_payload_authorize failed" ));
-      }
-      fd_ed25519_sign( ctx->out[ in_idx ].data, ctx->_data, 32UL, ctx->public_key, ctx->private_key, ctx->sha512 );
-      break;
-    }
-    case FD_KEYGUARD_ROLE_TLS: {
-      if( FD_UNLIKELY( !fd_keyguard_payload_authorize( ctx->_data, 130UL, FD_KEYGUARD_ROLE_TLS ) ) ) {
-        FD_LOG_EMERG(( "fd_keyguard_payload_authorize failed" ));
-      }
-      fd_ed25519_sign( ctx->out[ in_idx ].data, ctx->_data, 130UL, ctx->public_key, ctx->private_key, ctx->sha512 );
-      break;
-    }
-    case FD_KEYGUARD_ROLE_GOSSIP: {
-      if( FD_UNLIKELY( !fd_keyguard_payload_authorize( ctx->_data, *opt_sz, FD_KEYGUARD_ROLE_GOSSIP ) ) ) {
-        FD_LOG_EMERG(( "fd_keyguard_payload_authorize failed" ));
-      }
-      if ( fd_keyguard_payload_matches_ping_msg( ctx->_data, *opt_sz ) ) {
-        /* Gossip tile sends the sh256 pre-image for ping/pong msgs. */
-        uchar hash[32];
-        fd_sha256_hash( ctx->_data, *opt_sz, hash );
-
-        fd_ed25519_sign( ctx->out[ in_idx ].data, hash, 32UL, ctx->public_key, ctx->private_key, ctx->sha512 );
-      } else {
-        fd_ed25519_sign( ctx->out[ in_idx ].data, ctx->_data, *opt_sz, ctx->public_key, ctx->private_key, ctx->sha512 );
-      }
-
-      break;
-    }
-    case FD_KEYGUARD_ROLE_REPAIR: {
-      if( FD_UNLIKELY( !fd_keyguard_payload_authorize( ctx->_data, *opt_sz, FD_KEYGUARD_ROLE_REPAIR ) ) ) {
-        FD_LOG_EMERG(( "fd_keyguard_payload_authorize failed" ));
-      }
-      fd_ed25519_sign( ctx->out[ in_idx ].data, ctx->_data, *opt_sz, ctx->public_key, ctx->private_key, ctx->sha512 );
-      break;
-    }
-    default:
-      FD_LOG_CRIT(( "unexpected link role %lu", ctx->in_role[ in_idx ] ));
+  if( FD_UNLIKELY( !fd_keyguard_payload_authorize( ctx->_data, sz, role, sign_type ) ) ) {
+    FD_LOG_HEXDUMP_WARNING(( "failed signing request", ctx->_data, sz ));
```

Comment:
I don't like adding this log because it adds a potential vector for the other process to try and corrupt (eg, passing a lot of data and overflowing a log buffer or something).

Obviously that probably isn't possible, but keeping these vectors minimized is important for the signer, printing user provided buffers doesn't seem smart here.